### ISSUE TITLE
Support for frames/iframes

### ISSUE THREAD

**Issue by aik099** [qa-tools/qa-tools#116]
Idea is to allow specifying frame/iframe name where element is located so that when somebody wants to access it an automatic switch to that frame happens and then switch back to original window/frame happens.

Possible implementations:
* have inheritable `@frame('test-name')` or `@frame('name' => 'test-name')` annotation 
* have `frame` parameter for `@find-by` annotation (e.g. `@find-by('id' => 'test-id', 'frame' => 'test-name')`)

When used then:
* current frame will be changed using `switchToFrame` method of Mink
* a search will be made for an element
* current frame is changed back to what it was

If was used on element container (sub-class of `AbstractElementContainer`, that was called `HtmlElement` before), then there is no need to specify frame/iframe on each element in this container because they'll be searched within container located in another frame/iframe anyway.

__Egde cages__:

* nested framesets - I think Mink should be handling that already
* frames/iframes in another window/tab - I think it's out of library scope because user can switch window before doing element search

**Comment by evangelion1204** [comment]
`@frame` sounds better to me else `@find-by` might get too complex and an extra annotation is easier to spot.

**Comment by evangelion1204** [comment]
Just thinking about a possible implementation should we either enhance the `DefaultElementLocator` or create a new `FrameElementLocator`? The Factory could either return a Default or Frame depending on `Property`

**Comment by aik099** [comment]
The default element locator can handle that. Nobody really restricts one element locator to operate only on one annotation.

**Comment by evangelion1204** [comment]
In that case we need to access either the session or the driver itself to switch to Frame. Either we put it as port of the `Locator` or as part of the `ISearchContext`.

**Comment by aik099** [comment]
No need. The search context is a real element (if I'm not mistaken), not a proxy. When we switch a frame and find an element there, then the returned element reference (inside Mink) contains (that needs to be tested) the frame where element was found. So even if you switch the frame, then element should still act as `being inside a frame` and all subsequent `findAll` calls from locator should honor that.

**Comment by evangelion1204** [comment]
I tested it, the element is found. But accessing it and changing the value will force selenium to search for the element again to alter the value.

**Comment by aik099** [comment]
You're correct. Mink only retains XPATH of element found via the Driver (e.g. Selenium, Sahi). Then this xpath is used to perform all further actions (even finding sub-elements).

To make this happen we need to:

1. store frame handle (driver specific frame identifier) next to WebElement
2. before every operation switch frame to needed one (if they differ)
3. after every operation switch frame back (if needed)

Doesn't sound like an easy task. Let's postpone this for now.

**Comment by aik099** [comment]
Interesting idea posted here: https://github.com/yandex-qatools/htmlelements/issues/56#issuecomment-190629472

So we switch to frame once > do the stuff > switch back. Since we do it outside of the elements no need to switch back/forth on each element query.
